// ka-table/Components/CellEditorBoolean/CellEditorBoolean.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var actionCreators_1 = require("../../actionCreators");
var defaultOptions_1 = __importDefault(require("../../defaultOptions"));
var CommonUtils_1 = require("../../Utils/CommonUtils");
var ComponentUtils_1 = require("../../Utils/ComponentUtils");
var CellEditorBoolean = function (props) {
    var _a;
    var column = props.column, dispatch = props.dispatch, value = props.value, rowKeyValue = props.rowKeyValue, autoFocus = props.autoFocus, childComponents = props.childComponents;
    var _b = ComponentUtils_1.getElementCustomization({
        className: "" + defaultOptions_1.default.css.checkbox,
        autoFocus: autoFocus,
        type: 'checkbox',
        checked: value || false,
        onChange: function (event) { return dispatch(actionCreators_1.updateCellValue(rowKeyValue, column.key, event.currentTarget.checked)); },
        onBlur: function () { return dispatch(actionCreators_1.closeEditor(rowKeyValue, column.key)); }
    }, props, (_a = childComponents) === null || _a === void 0 ? void 0 : _a.cellEditorInput), elementAttributes = _b.elementAttributes, content = _b.content;
    return (content ||
        (react_1.default.createElement("input", __assign({ ref: function (elem) { return elem && (elem.indeterminate = CommonUtils_1.isEmpty(value)); } }, elementAttributes))));
};
exports.default = CellEditorBoolean;
