// ka-table/Utils/CellResizeUtils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HeadCellResizeStateAction = 'HeadCellResizeStateAction';
exports.isCellResizeShown = function (isResizable, columnResizing) { return !!((isResizable !== false) && (columnResizing || isResizable)); };
exports.getMouseMove = function (currentWidth, minWidth, startX, dispatch) { return function (event) {
    var newWidth = event.screenX - startX;
    if (newWidth !== currentWidth) {
        newWidth = exports.getValidatedWidth(newWidth, minWidth);
        dispatch({ type: exports.HeadCellResizeStateAction, width: newWidth });
    }
}; };
exports.getValidatedWidth = function (newWidth, minWidth) {
    if (newWidth < minWidth) {
        return minWidth;
    }
    return newWidth;
};
exports.isNumberWidth = function (width) { return width && typeof width === 'number'; };
exports.getMinWidth = function (style) {
    var minWidth = 20;
    var styleMinWidth = style && style.minWidth;
    if (exports.isNumberWidth(styleMinWidth)) {
        minWidth = styleMinWidth;
    }
    return minWidth;
};
exports.headCellDispatchWrapper = function (setWidth, dispatch) { return function (action) {
    if (action.type === exports.HeadCellResizeStateAction) {
        setWidth(action.width);
    }
    else {
        dispatch(action);
    }
}; };
